### ISSUE TITLE
VMSingle labels are copied to PVC

### ISSUE THREAD

**Issue by amkartashov** [VictoriaMetrics/operator#432]
See https://github.com/VictoriaMetrics/operator/issues/90 - I can't reopen it.

Details: https://github.com/VictoriaMetrics/operator/issues/90#issuecomment-1045888479

**Comment by amkartashov** [comment]
Similar issues in other operator: https://github.com/strimzi/strimzi-kafka-operator/issues/2107

**Comment by f41gh7** [comment]
WDYT, could it be fixed by automatically adding additional annotations to the generated PVC:

```yaml

metadata:
  annotations:
    argocd.argoproj.io/compare-options: IgnoreExtraneous
    argocd.argoproj.io/sync-options: Prune=false
```

 I'm not so familiar with ArgoCD, but label and annotation propagation is very useful for me.

**Comment by amkartashov** [comment]
@f41gh7 I'm not sure how would I add metadata to PVC which is created and controller by operator :)

This is not how it works in prometheus-operator, this is not how it works in Deployment controller. Also, this might break other tools, because you copy "managed-by" labels as well. F.e. Helm might be confused by it.

> but label and annotation propagation is very useful for me

Not sure what profit it gives.

**Comment by f41gh7** [comment]
> @f41gh7 I'm not sure how would I add metadata to PVC which is created and controller by operator :)
> 
> This is not how it works in prometheus-operator, this is not how it works in Deployment controller. Also, this might break other tools, because you copy "managed-by" labels as well. F.e. Helm might be confused by it.
> 
> > but label and annotation propagation is very useful for me
> 
> Not sure what profit it gives.

 Hm, make sense.

 I think, that it should be changed - it's better to add selector labels to the PVC. It allows to identify created PVC and discover it if needed. Managed labels wouldn't be transfered to the PVC.

 selector labels:
```
 "app.kubernetes.io/name":      "vmsingle",
"app.kubernetes.io/instance":  cr.Name,
"app.kubernetes.io/component": "monitoring",
"managed-by":                  "vm-operator",
```

**Comment by f41gh7** [comment]
Well, it seems like it's better to introduce new option - `storageMetadata` for `VMSingle`, it allows to configure labels and annotations to the created PVC. Hope it must solve current issue with well-known labels.

**Comment by amkartashov** [comment]
@f41gh7 yep, this would be perfect solution.

**Comment by f41gh7** [comment]
Will be added at the next release.

**Comment by f41gh7** [comment]
Changes was included to the v0.24.0 release. Feel free to re-open the issue if it doesn't solve it.
